### PR TITLE
Ss extra sections

### PR DESCRIPTION
**Pull request by Pattrigue**

### CODE REVIEW

File: Report/sections/2_design/semantics.tex [hunk 1]
```diff
@@ -66,6 +66,14 @@ \subsection*{Type rules in \dazel{}}
 \ref{item:tiletype} is an important type: it is the type behind every tile in the generated games.
 \ref{item:arraytype} means that every type in the list will have a corresponding array type. This is to say that arrays are parametric - for example, an array could be an array of numbers.
 
+\subsection*{Dynamic and static binding}
+A variable must first be bound to a data type before it can be used in the program.
```
Comment (chhoumann) [inline Report/sections/2_design/semantics.tex:70]: We don't explicitly declare types, which I think this sentence implies. When a variable is assigned, it receives the type that the expression evaluates to.

File: Report/sections/2_design/semantics.tex [hunk 2]
```diff
@@ -66,6 +66,14 @@ \subsection*{Type rules in \dazel{}}
 \ref{item:tiletype} is an important type: it is the type behind every tile in the generated games.
 \ref{item:arraytype} means that every type in the list will have a corresponding array type. This is to say that arrays are parametric - for example, an array could be an array of numbers.
 
+\subsection*{Dynamic and static binding}
+A variable must first be bound to a data type before it can be used in the program. 
+This is called a binding and, depending on the language, it can either be dynamic or static.
+\textbf{Dynamic binding} is when the type of a variable is determined at runtime. 
+\textbf{Static binding} is when the type of a variable can be determined during compilation as all the necessary information is present in the source code \cite{sebesta_concepts_2016}.
+
+\dazel{} uses static binding as types are determined while the source code is compiled to API calls. When a variable is initially declared, it is assigned a type before the code is run.
```
Comment (chhoumann) [inline Report/sections/2_design/semantics.tex:75]: We shouldn't write 'compiled' when we're building an interpreter. Typechecking can be mentioned here - that's where we determine which types variables have.

File: Report/sections/2_design/expressions.tex [hunk 3]
```diff
@@ -1 +1,37 @@
...
+\subsubsection*{Boolean expressions}
+Boolean expressions are constructed using different mathematical logic operators such as the logical \texttt{AND}, \texttt{OR} and \texttt{EQUALS}. 
+When evaluated, a whole expression is either true or false. 
+This is then often combined with statements such as \texttt{if-then-else} statements to achieve a conditional program flow - if the expression evaluates to true, the code in the \texttt{then} block is executed, and if it evaluates to false, code in the \texttt{else} block is executed instead\cite{sebesta_concepts_2016}.
+
+\subsubsection*{Expressions in \dazel{}}
+\dazel{} features a variety of binary operators to support arithmetic as well as unary operations for boolean expressions.
+It also has an operator evalation order identical to mathematics, as well as a left to right associativity. This helps prevent confusion for beginner programmers, 
+and also maintains the standard that most experienced programmers are accustomed to.
+To accomodate evalation of expressions \dazel{} also has the most common relational and logic operators as can be seen in \tabref{table:operatorsymbols}.
```
Comment (chhoumann) [inline Report/sections/2_design/expressions.tex]: This entire section reads as if it were written off from Sebasta - or, at least, someone's notes on the book. The introduction is weird and does not say anything. We don't really deliver on the promise to explain the relevance of the types to Dazel - we just explain what the types are.

There are too many subsections.

My suggestion: reduce this section to the very last few lines. Expand upon those. Line 34 mentions a 'variety of binary operators'. Refer to the table here. Show what Dazel has.

```suggestion
\dazel{} supports a variety of binary operators for arithmetic expressions, as well as unary operations for boolean expressions. These operators can be seen in \tabref{table:operatorsymbols}.
Operator evalation order in \dazel{} is identical to that seen in mathematics.
```

We have no data to back up the claim that "_This helps prevent confusion for beginner programmers, 
and also maintains the standard that most experienced programmers are accustomed to._" - unless, of course, I've missed something. In that case, please cite it here.

Perhaps move the table containing the operators (and the section that introduces it) closer to this section. Or merge them.

**Review by chhoumann — CHANGES_REQUESTED** [review]
Please see requested changes and attached comments.

File: Report/sections/2_design/semantics.tex (hunk 1 above)
Comment (Pattrigue) [reply Report/sections/2_design/semantics.tex:70]: This is the explanation of the general concept, not how it works in Dazel

File: Report/sections/2_design/semantics.tex (hunk 2 above)
Comment (Pattrigue) [reply Report/sections/2_design/semantics.tex:75]: The code still gets compiled into some intermediate code though. The intermediate code is then what is interpreted. 
Also type determination is mentioned already.

File: Report/sections/2_design/expressions.tex (hunk 3 above)
Comment (Pattrigue) [reply Report/sections/2_design/expressions.tex]: We need to introduce all the important concepts and theory before we can argue why Dazel works the way it does. Similarly to other sections, such as the one about scopes, we first describe the concepts, then relate them to Dazel, as was requested by the supervisor.

**Review by chhoumann — APPROVED** [review]

**Pull request merged**